inst/examples/build-a-box/R/mod-value-box-ui.R

```diff
@@ -0,0 +1,255 @@
+`%||%` <- function(x, y) if (is.null(x)) y else x
+
+icon_list <- list(
+  "bsicons" = setNames(
+    paste0("bsicons::", bsicons:::icon_info$name),
+    bsicons:::icon_info$name
+  ),
+  "fontawesome" = setNames(
+    paste0("fontawesome::", fontawesome::fa_metadata()$icon_names),
+    fontawesome::fa_metadata()$icon_names
+  )
+)
+
+resolve_icon <- function(name) {
+  icon <- strsplit(name, "::", fixed = TRUE)[[1]]
+  icon_fn <- switch(
+    icon[1],
+    bsicons = {
+      ns <- "bsicons"
+      fn <- "bs_icon"
+    },
+    {
+      ns <- "fontawesome"
+      fn <- "fa_i"
+    }
+  )
+  rlang::call2(.ns = ns, fn, icon[2])
+}
+
+ui_value_box_options <- function(id) {
+  ns <- NS(id)
+
+  conditional_panel <- function(condition, ...) {
+    conditionalPanel(condition, ..., ns = ns)
+  }
+
+  init <- random_title_value()
+
+  list(
+    "title_value" = list(
+      textInput(ns("title"), "Title", init$title),
+      textInput(ns("value"), "Value", init$value),
+      tagAppendAttributes(
+        textAreaInput(ns("extra"), "Extra text (markdown allowed)", ""),
+        class = "input-text-code"
+      ),
+      actionButton(ns("random_stat"), "Random stat")
+    ),
+    "theme_opts" = list(
+      input_switch(ns("use_theme"), "Use theme", TRUE),
+      conditional_panel(
+        "input.use_theme",
+        ui_selextra(ns("theme"), "Theme")
+      ),
+      conditional_panel(
+        "!input.use_theme",
+        colourpicker::colourInput(ns("foreground"), "Foreground", value = "#000000"),
+        colourpicker::colourInput(ns("background"), "Background", value = "#FFFFFF")
+      ),
+      input_switch(ns("full_screen"), "Allow full screen", value = FALSE),
+      input_switch(ns("fill"), "Fill vertical space", value = TRUE),
+      input_switch(ns("fixed_height"), "Fixed height", value = FALSE),
```

Comment:
It kind of feels like these could/should be global controls

Reply:
Full screen follows the plot/icon choice and the other two could definitely be global options... but I think they'd clutter the initial UI without adding much value there.